geth/process.py

```diff
@@ -318,7 +318,8 @@ def __init__(
             base_dir = get_default_base_dir()
 
         self._data_dir = get_chain_data_dir(base_dir, chain_name)
-        geth_kwargs = construct_test_chain_kwargs(data_dir=self.data_dir, **overrides)
+        overrides["data_dir"] = self._data_dir
+        geth_kwargs = construct_test_chain_kwargs(**overrides)
         validate_geth_kwargs(geth_kwargs)
 
         # ensure that an account is present
```

Comment:
`data_dir` is a property function inherited from the base class. It should be retrieved with `self.data_dir` at least. Since it's only set the once here I left it using the underscored, but maybe I should make it a setter-decorated function call?

Reply:
On more thinking, no, the `data_dir` needs to be passed to the geth process during init, so I don't see that it would be needful to update it without also needing to start a new geth process. Leaving the only assignment going to `self._data_dir` and all retrievals from `self.data_dir`.

Reply:
Ah, they are meant to be different. Thanks for the clarification!